Client/Frontend/Browser/BrowserViewController/ProductNotifications/BlockingSummaryDataSource.swift

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import Shared
+import BraveShared
+
+private let log = Logger.browserLogger
+
+// MARK: - BlockingSummary
+
+struct BlockingSummary: Codable {
+    
+    // MARK: CodingKeys
+    
+    enum CodingKeys: String, CodingKey {
+        case site
+        case savings
+        case numchildpages = "numofchildpages"
+        case childsavings = "avgchildsavings"
+        case sitesavings = "avgsitesavings"
+        case totalsavings
+        case avgsavings
+    }
+    
+    // MARK: Internal
+    
+    let site: String
+    let savings: Int
+    let numchildpages: Int
+    let childsavings: Double
+    let sitesavings: Double
+    let totalsavings: Double
+    let avgsavings: String
+    
+    // MARK: Lifecycle
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        
+        site = try container.decode(String.self, forKey: .site)
+        savings = try container.decode(Int.self, forKey: .savings)
+        numchildpages = try container.decode(Int.self, forKey: .numchildpages)
+        childsavings = try container.decode(Double.self, forKey: .childsavings)
+        sitesavings = try container.decode(Double.self, forKey: .sitesavings)
+        totalsavings = try container.decode(Double.self, forKey: .totalsavings)
+        
+        let unformattedSaving = try container.decode(Double.self, forKey: .avgsavings)
+        avgsavings = String(format: "%.2f", unformattedSaving)
+    }
+}
+
+// MARK: - BlockingSummaryDataSource
+
+class BlockingSummaryDataSource {
+    
+    // MARK: Lifecycle
+    
+    init() {
+        blockingSummaryList = fetchBlockingSummaryObjects()
+    }
+    
+    // MARK: Internal
+    
+    func fetchDomainFetchedSiteSavings(_ url: URL) -> String? {
```

Comment:
Would be good to unit test it, to make sure it parses domains correctly,
Maybe add a mock json file to the test case then run this function through few domains, and verify it does not show on non-eligible domains

Reply:
Detailed Tests are added https://github.com/brave/brave-ios/blob/800aaab49c7a7cc020ba25025c55908906d7d501/ClientTests/BlockSummaryTests.swift

The blocking data source is changed to support testability.